### PR TITLE
Iron 386 replace all system drawing with iron drawing

### PR DESCRIPTION
**Pull request by mee-ironsoftware**
# Features Added
## AnyBitmap
- Scan0 to gets the address of the first pixel data of the AnyBitmap object
- Stride to gets the stride width (also called scan width) of the AnyBitmap object
 MimeType to gets MIMEType of the AnyBitmap object including image/bmp, image/jpeg, image/png, image/gif, image/tiff, image/webp, and image/unknown
- GetImageFormat to gets ImageFormat of the AnyBitmap object
- HorizontalResolution, and VerticalResolution to gets the resolution of the image in x, y direction
## Color
- Add override FromArgb(alpha, BaseColor)
- Add compare Color including ==, !=, and Equals
- Create Color from name

# Bug Fixed
## AnyBitmap
- Fixed loss transparency when convert System.Drawing.Bitmap to AnyBitmap
- Fixed loss transparency when convert System.Drawing.Image to AnyBitmap
- Fixed loss transparency when convert SixLabors.ImageSharp.Image to AnyBitmap

### CODE REVIEW

File: IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs [hunk 1]
```diff
@@ -601,7 +699,12 @@ public static AnyBitmap CreateMultiFrameGif(IEnumerable<AnyBitmap> images)
             {
                 using (var memoryStream = new System.IO.MemoryStream())
                 {
-                    Image.Save(memoryStream, new SixLabors.ImageSharp.Formats.Bmp.BmpEncoder());
+                    // BMP transparency support only 32 bits
```
Comment (iron-production-team) [inline IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs:702]: says only support 32 bits per pixel in comments, but implemented as 24

File: NuGet/IronSoftware.Drawing.nuspec [hunk 2]
```diff
@@ -36,21 +36,30 @@ Supports:
...
+    - GetImageFormat to gets ImageFormat of the AnyBitmap object
+    - HorizontalResolution, and VerticalResolution to gets the resolution of the image in x, y direction
+  * Color
+    - Add override FromArgb(alpha, BaseColor)
+    - Add compare Color including ==, !=, and Equals
+    - Create Color from name
+
+Bug Fixed
+  * AnyBitmap
+    - Fixed loss transparency when convert System.Drawing.Bitmap to AnyBitmap
```
Comment (michael-ironsoftware) [inline NuGet/IronSoftware.Drawing.nuspec:53]: Would change `convert` to `converting` for these.

File: IronSoftware.Drawing/IronSoftware.Drawing.Common/KnownColors.cs [hunk 3]
```diff
@@ -0,0 +1,436 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace IronSoftware.Drawing
+{
+    internal class KnownColors
+    {
+        static internal uint[] ArgbValues = new uint[] {
```
Comment (iron-production-team) [inline IronSoftware.Drawing/IronSoftware.Drawing.Common/KnownColors.cs:8]: Would `Dictionary <KnownColor, uint>` be more readable and harder to make mistakes?

File: IronSoftware.Drawing/IronSoftware.Drawing.Common/KnownColors.cs [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+            uint argb = (uint)c.ToArgb();
+            for (int i = 0; i < KnownColors.ArgbValues.Length; i++)
+            {
+                if (argb == KnownColors.ArgbValues[i])
+                    return KnownColors.FromKnownColor((KnownColor)i);
+            }
+            return Color.Empty;
+        }
+
+        // When this method is called, we teach any new color(s) to the Color class
```
Comment (iron-production-team) [inline IronSoftware.Drawing/IronSoftware.Drawing.Common/KnownColors.cs:429]: Cam hide from intellisense if will not be used by developers

File: IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs (hunk 1 above)
Comment (mee-ironsoftware) [reply IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs:702]: This is just the note.
When loading Image with SixLabors.ImageSharp.Image<SixLabors.ImageSharp.PixelFormats.Rgb24> it'll loss the transparency at first.
Maybe the better way is just removed the comment that may confusion.
What do you think?

File: IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs [hunk 5]
```diff
@@ -590,6 +610,84 @@ public static AnyBitmap CreateMultiFrameGif(IEnumerable<AnyBitmap> images)
...
+                case "image/png": return ImageFormat.Png;
+                case "image/webp": return ImageFormat.Webp;
+                case "image/vnd.microsoft.icon": return ImageFormat.Icon;
+
+                default: return ImageFormat.Bmp;
+            }
+        }
+
+        /// <summary>
+        /// Gets the resolution of the image in x- direction.
```
Comment (michael-ironsoftware) [inline IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs:668]: `x-direction`

File: IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs [hunk 6]
```diff
@@ -590,6 +610,84 @@ public static AnyBitmap CreateMultiFrameGif(IEnumerable<AnyBitmap> images)
...
+        public double? HorizontalResolution
+        {
+            get
+            {
+                return Image?.Metadata.HorizontalResolution ?? null;
+            }
+        }
+
+        /// <summary>
+        /// Gets the resolution of the image in y- direction.
```
Comment (michael-ironsoftware) [inline IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs:680]: `y-direction`

**Review by michael-ironsoftware — APPROVED** [review]
Looks great. Thanks.

**Comment by iron-production-team** [comment]
This PR looks good to me

Are there any specific files or parts you wish me to look over or review in more detail?

File: IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs [hunk 7]
```diff
@@ -590,6 +610,84 @@ public static AnyBitmap CreateMultiFrameGif(IEnumerable<AnyBitmap> images)
...
+        public IntPtr Scan0
+        {
+            get
+            {
+                return GetFirstPixelData();
+            }
+        }
+
+        /// <summary>
+        /// The MIMEType must be one of the following:
```
Comment (iron-production-team) [inline IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs:637]: Returns the http mime type of the image.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types

**Pull request merged**